### PR TITLE
feat(Application): Add crud

### PR DESCRIPTION
**Pull request by Gio85**
Add crud

feat(Application): Add crud

Add crud

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -1,6 +1,13 @@
 export enum HOST {
   // TODO @Team remember to change this to your localhost for testing
-  API = 'https://ossia-studio-api.aimusic.services',
+  API = 'http://localhost:8081',
```
Comment (Dextorr) [inline src/types.ts:3]: API address?

Comment (Gio85) [reply src/types.ts:3]: yeah I saw it..

File: src/api/client/endpoints/application.ts [hunk 2]
```diff
@@ -1,14 +1,46 @@
...
+import { ENDPOINT, IApiResponse, IApplication, IHttpClient, IQueryParameters } from '../../../types'
+import { stringifyQueryObject } from '../../../utils'
 
 export class ApplicationEndpoint {
   public path = `${ENDPOINT.APPLICATIONS}`
   constructor(protected readonly client: IHttpClient) {}
 
   /**
-   * This method will log you in with your application credentials
+   * This method will allow you to create the application
```
Comment (useless-stuff) [inline src/api/client/endpoints/application.ts:9]: an application

File: src/api/client/endpoints/application.ts [hunk 3]
```diff
@@ -1,14 +1,46 @@
...
+   * This method will return the application model
+   * @param applicationId
+   */
+  public async get(applicationId: string): Promise<IApiResponse<IApplication>> {
+    return this.client.get<IApplication>(`${this.path}/${applicationId}`)
+  }
+
+  /**
+   * This method will return the application list
+   * Please bare in mind that this method will require to have an admin token in order to receive the data
```
Comment (useless-stuff) [inline src/api/client/endpoints/application.ts:26]: Please bear in mind that this method requires a specific permission

File: src/api/client/endpoints/application.ts [hunk 4]
```diff
@@ -1,14 +1,46 @@
...
+    if (parameters) {
+      const queryParameters = stringifyQueryObject(parameters)
+      return this.client.get<IApplication[]>(`${this.path}?${queryParameters}`)
+    }
+    return this.client.get<IApplication[]>(this.path)
+  }
+
+  /**
+   * This method will update the given application
+   * Please bare in mind that this method will require to have an admin token in order to update the model
```
Comment (useless-stuff) [inline src/api/client/endpoints/application.ts:40]: Admin is a role, you could have just permission set to your app account and be able to fetch the data

File: src/api/client/endpoints/auth.ts [hunk 5]
```diff
@@ -12,4 +15,28 @@ export class AuthEndpoint implements IAuthEndpoint {
     // TODO add validation from credentials
```
Comment (useless-stuff) [inline src/api/client/endpoints/auth.ts:15]: ? do we have a story for this TODO:? if not and we don't care remove it

File: src/api/client/endpoints/auth.ts [hunk 6]
```diff
@@ -12,4 +15,28 @@ export class AuthEndpoint implements IAuthEndpoint {
...
     return this.client.post<IAuthResponse>(this.path, payload)
   }
+
+  public setApplicationToken(token: string | null): void {
+    if (token) {
```
Comment (useless-stuff) [inline src/api/client/endpoints/auth.ts:20]: () ? :

File: src/types.ts (hunk 1 above)
Comment (useless-stuff) [reply src/types.ts:3]: this should be set in the client In a way that allows us to have an env var but fallback to production if the env file is not set, something like:

```
const host = process.env.API_HOST || HOST.API
```

File: src/api/client/endpoints/auth.ts [hunk 7]
```diff
@@ -12,4 +15,28 @@ export class AuthEndpoint implements IAuthEndpoint {
...
+
+  protected setApplicationId(userId: string): void {
+    this.applicationId = userId
+  }
+
+  public getApplicationId(): string {
+    return this.applicationId
+  }
+
+  public getDecodedToken(token: string | null): IJwtTokenPayload {
```
Comment (useless-stuff) [inline src/api/client/endpoints/auth.ts:36]: do we care to decode the token client side?

Comment (Gio85) [reply src/api/client/endpoints/auth.ts:36]: yes in order to set the AppId once the application has been authenticated

File: src/api/client/endpoints/application.ts (hunk 4 above)
Comment (Gio85) [reply src/api/client/endpoints/application.ts:40]: this one will require to change the BE or at least to change the way we create the app maybe setting permission in a proper way now..

**Pull request merged**